scripts/zones/Nyzul_Isle/instances/nyzul_isle_investigation.lua

```diff
@@ -0,0 +1,182 @@
+-----------------------------------
+--
+-- Assault 51 Nyzul Isle Investigation
+--
+-----------------------------------
+local ID = require("scripts/zones/Nyzul_Isle/IDs")
+require("scripts/globals/instance")
+require("scripts/globals/keyitems")
+require("scripts/globals/status")
+require("scripts/zones/Nyzul_Isle/globals")
+-----------------------------------
+
+function afterInstanceRegister(player)
+    local instance = player:getInstance()
+    player:messageSpecial(ID.text.TIME_TO_COMPLETE, instance:getTimeLimit())
+end
+
+function onInstanceCreated(instance)
+end
+
+function onInstanceTimeUpdate(instance, elapsed)
+    updateInstanceTime(instance, elapsed, ID.text)
+end
+
+function onInstanceFailure(instance)
+    local chars = instance:getChars()
+
+    for i,v in ipairs(chars) do
+        v:messageSpecial(ID.text.MISSION_FAILED,10,10)
+        v:startEvent(1)
+    end
+end
+
+function onInstanceProgressUpdate(instance, progress)
+    local chars = instance:getChars()
+    local stage = instance:getStage()
+
+    if stage == (nyzul.objective.FREE_FLOOR or stage == nyzul.objective.ELIMINATE_ENEMY_LEADER or stage == nyzul.objective.ELIMINATE_SPECIFIED_ENEMY) and progress == 15 then
```

Comment:
Im thinking of breaking it down to a magic number for all floors. Im just not there yet. Plan is i add a counter to every mob that spawns and set progress say 15 - (however mobs spawned). So sometines it could be 15 - 6 then kill the 9 amd rune light ect.  Im just nit at that point yet to make it a reality but i have a thought in place